### PR TITLE
Fastapi Setup

### PR DESCRIPTION
**Pull request by smartwhatt**
# Features
- add Fastapi
- add types utils

### CODE REVIEW

File: src/utils/types/student_teacher/people.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    Mrs = "นาง"
+    Miss = "นางสาว"
+
+class EnglishPrefix(str, Enum):
+    """
+    English prefix for name
+    """
+    Mr = "Mr."
+    Mrs = "Mrs."
+    Miss = "Miss."
```
Comment (SiravitPhokeed) [inline src/utils/types/student_teacher/people.py:23]: What about “Master”?

File: src/utils/types/atk/atk.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from pydantic import BaseModel
+from typing import List, Optional
+from enum import Enum
+
+from student_teacher.people import People
+
+class MethodEnum(str, Enum):
+    oral = "oral"
+    injection = "injection"
```
Comment (SiravitPhokeed) [inline src/utils/types/atk/atk.py]: I think what you meant was “nasal“, I don’t think you can get jabbed for an ATK test.

File: src/utils/types/classroom/classroom.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from enum import Enum
+from pydantic import BaseModel
+from typing import List, Optional
+
+from student_teacher.teacher import Teacher
+from student_teacher.student import Student
+
+# from subject.subjects import Subject
```
Comment (SiravitPhokeed) [inline src/utils/types/classroom/classroom.py]: I don’t think we would need to have Subject be tied to Classroom so you can probably remove this.

File: src/utils/types/student_teacher/contacts.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+class ContactType(str, Enum):
+    """
+    Contact type
+    """
+    Phone = "phone"
+    Email = "email"
+    Facebook = "facebook"
+    Line = "line"
+    Other = "other"
```
Comment (SiravitPhokeed) [inline src/utils/types/student_teacher/contacts.py]: I thought you were going to have Instagram and Website too.

File: src/utils/types/student_teacher/schema.sql [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+CREATE TABLE Contact
+(
+    id INTEGER PRIMARY KEY,
+    type INTEGER NOT NULL REFERENCES ContactType (id),
+    name TEXT NOT NULL,
+    value TEXT NOT NULL,
+);
+
+CREATE TABLE People
```
Comment (SiravitPhokeed) [inline src/utils/types/student_teacher/schema.sql]: “Person” would probably make more sense.

**Review by SiravitPhokeed — COMMENTED** [review]
The formatting is, in my opinion, pretty wack.

**Review by SiravitPhokeed — APPROVED** [review]
Looks great to me!

File: src/utils/types/auth/user.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+from student_teacher.student import Student
+from student_teacher.teacher import Teacher
+
+
+from pydantic import BaseModel, validator
```
Comment (Temp9699) [inline src/utils/types/auth/user.py]: "validator" never used in the file

File: src/utils/types/student_teacher/people.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+from pydantic import BaseModel
+from typing import List, Optional
+from datetime import datetime, date
```
Comment (Temp9699) [inline src/utils/types/student_teacher/people.py:3]: "datetime" never used in the file

**Review by Temp9699 — APPROVED** [review]
don't import things you don't use

File: src/utils/types/student_teacher/teacher.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
+from .person import Person
```
Comment (ImSadudee) [inline src/utils/types/student_teacher/teacher.py]: .person inconsistent with other imports

File: src/utils/types/subject/subject_group.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
+from lib2to3.pgen2.token import OP
```
Comment (smartwhatt) [inline src/utils/types/subject/subject_group.py]: Why did I add this

**Review by ImSadudee — APPROVED** [review]
Please review comment before merging

**Pull request merged**